Tests/NIOTests/EventLoopTest.swift

```diff
@@ -613,6 +613,40 @@ public final class EventLoopTest : XCTestCase {
         XCTAssertFalse(channel.isActive)
     }
 
+    public func testFlatSubmitCloseFutureNotifiedBeforeUnblock() throws {
+        class AssertHandler: ChannelInboundHandler {
```

Comment:
this test case tests the new `flatSubmit` functionality but also a quite a bit of unrelated other unrelated stuff. I think we could reduce this test case to only the things that are actually needed such as:

- a test case that `flatSubmit`s a successful computation to `EmbeddedEventLoop` 
- a test case that `flatSubmit`s a failing computation to `EmbeddedEventLoop`
- a test case that submits a computation to a `MultiThreadedEventLoopGroup` that is already shut down.

as an example for the first case:

```
func testSuccessfulFlatSubmit() {
    let el = EmbeddedEventLoop()
    XCTAssertNoThrow(XCTAssertEqual(1, try el.flatSubmit {
        el.makeSucceededFuture(1)
    }.wait()))
}
```

Reply:
Agreed. Thanks for the example. I was unfamiliar with the EmbeddedEventLoop, so that was helpful.